src/Model.elm

```diff
@@ -179,17 +179,22 @@ update msg model =
                     ( { model | error = Just (toString err) }, Cmd.none )
 
         DeleteRecord recordId ->
-            ( model, deleteRecord recordId )
+            ( model, deleteRecord model recordId )
 
         DeleteRecordResponse response ->
             case response of
                 Ok data ->
-                    ( { model
-                        | records = removeRecordFromList data model.records
-                        , error = Nothing
-                      }
-                    , fetchRecords model
-                    )
+                    case (decodeValue decodeRecord data) of
+                        Err msg ->
+                            ( { model | error = Just msg }, Cmd.none )
+
+                        Ok record ->
+                            ( { model
+                                | records = removeRecordFromList record model.records
+                                , error = Nothing
+                              }
+                            , fetchRecords model
+                            )
 
                 Err err ->
                     ( { model | error = Just (toString err) }, Cmd.none )
```

Comment:
It feels like there's a lot of boilerplate with handling errors in the different cases. Can we DRY them somehow, or refactor them using `Result.andThen`?